server/container_restore.go

```diff
@@ -331,6 +333,13 @@ func (s *Server) CRImportCheckpoint(
 			}
 		}
 
+		// Recursive Read-only (RRO) support requires the mount to be
+		// read-only and the mount propagation set to private.
+		if mount.RecursiveReadOnly {
+			mount.Readonly = true
+			mount.Propagation = types.MountPropagation_PROPAGATION_PRIVATE
+		}
+
```

Comment:
**To the reviewer:**

Hopefully, this is the right thing to do when restoring a container.